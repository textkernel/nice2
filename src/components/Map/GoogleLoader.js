import React from 'react';
import PropTypes from 'prop-types';
import { useLoadScript } from '@react-google-maps/api';
import { LoadingSpinner } from '../../index';

const GoogleLoader = props => {
    const { apiKey, loadErrorMessage, children } = props;
    const { isLoaded, loadError } = useLoadScript({ googleMapsApiKey: apiKey });

    if (loadError) {
        return <div>{loadErrorMessage}</div>;
    }

    return isLoaded ? (
        <React.Fragment>{children}</React.Fragment>
    ) : (
        <LoadingSpinner centerIn="parent" />
    );
};

GoogleLoader.displayName = 'GoogleLoader';

GoogleLoader.propTypes = {
    /** Google API key */
    apiKey: PropTypes.string.isRequired,
    /** Message to be shown if error occured during loading Google API */
    loadErrorMessage: PropTypes.string.isRequired,
    /** The map to be rendered */
    children: PropTypes.node.isRequired
};

export default GoogleLoader;
