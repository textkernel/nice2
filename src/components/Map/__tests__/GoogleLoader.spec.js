import React from 'react';
import toJson from 'enzyme-to-json';
import googleApi from '@react-google-maps/api';
import GoogleLoader from '../GoogleLoader';

const useLoadMock = jest.fn(() => ({ isLoaded: false, loadError: false }));
googleApi.useLoadScript = useLoadMock;

describe('<GoogleLoader> that wrapps a map and makes sure google API is loaded', () => {
    const API_KEY = 'AIzaSyA61aDjG48kvpwLhP-uE0sFSuHQ_j3EIwM';
    const mockErrorMessage = 'An error occured';
    it('should render a spinner while loading', () => {
        const wrapper = mount(
            <GoogleLoader loadErrorMessage={mockErrorMessage} apiKey={API_KEY}>
                <p>A map</p>
            </GoogleLoader>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('LoadingSpinner')).toHaveLength(1);
    });
    it('should render correctly when finished loading', () => {
        useLoadMock.mockImplementationOnce(() => ({ isLoaded: true, loadError: false }));
        const wrapper = mount(
            <GoogleLoader loadErrorMessage={mockErrorMessage} apiKey={API_KEY}>
                <div>A map</div>
            </GoogleLoader>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render error message if loading failed', () => {
        useLoadMock.mockImplementationOnce(() => ({ isLoaded: true, loadError: true }));
        const wrapper = mount(
            <GoogleLoader loadErrorMessage={mockErrorMessage} apiKey={API_KEY}>
                <div>A map</div>
            </GoogleLoader>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
