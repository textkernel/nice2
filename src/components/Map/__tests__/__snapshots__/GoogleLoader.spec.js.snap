// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleLoader> that wrapps a map and makes sure google API is loaded should render a spinner while loading 1`] = `
<GoogleLoader
  apiKey="AIzaSyA61aDjG48kvpwLhP-uE0sFSuHQ_j3EIwM"
  loadErrorMessage="An error occured"
>
  <LoadingSpinner
    centerIn="parent"
    context="brand"
    hidden={false}
    size={null}
  >
    <div
      aria-busy={true}
      aria-hidden={false}
      className="LoadingSpinner LoadingSpinner--centerIn_parent"
      role="status"
    >
      <svg
        className="LoadingSpinner__svg"
        style={null}
        viewBox={
          Array [
            0,
            0,
            44,
            44,
          ]
        }
      >
        <circle
          className="LoadingSpinner__path LoadingSpinner__path--context_brand"
          cx="22"
          cy="22"
          fill="none"
          r="20"
          strokeWidth="4"
        />
      </svg>
    </div>
  </LoadingSpinner>
</GoogleLoader>
`;

exports[`<GoogleLoader> that wrapps a map and makes sure google API is loaded should render correctly when finished loading 1`] = `
<GoogleLoader
  apiKey="AIzaSyA61aDjG48kvpwLhP-uE0sFSuHQ_j3EIwM"
  loadErrorMessage="An error occured"
>
  <div>
    A map
  </div>
</GoogleLoader>
`;

exports[`<GoogleLoader> that wrapps a map and makes sure google API is loaded should render error message if loading failed 1`] = `
<GoogleLoader
  apiKey="AIzaSyA61aDjG48kvpwLhP-uE0sFSuHQ_j3EIwM"
  loadErrorMessage="An error occured"
>
  <div>
    An error occured
  </div>
</GoogleLoader>
`;
